logging/logger.go

```diff
@@ -42,15 +42,43 @@ var (
 	defaultLoggerOnce sync.Once
 )
 
+// Factory is a factory to create loggers.
+type Factory struct {
+	Level, Mode string
+}
+
+// SetLevel sets the level for the logger factory.
+func (f *Factory) SetLevel(level string) {
```

Comment:
I know the fluent pattern isn't loved in golang, but this seems like the ideal place for it. Why isn't it liked?

Reply:
You mean like `WithLevel(level string) *Factory`?

Reply:
yeah

Reply:
I was actually trying to do that. However, it doesn't work very well with Go interface. E.g.

```go
type LoggerConfigurer interface {
  WithLevel(level string) LoggerConfigurer
}

type Factory struct {}

// This doesn't implement the interface above because of the return value type.
// But it's also kinda bad to return LoggerConfigurer instead.
func (f *Factory) WithLevel(level string) *Factory {}
```